packages/enzyme/src/Utils.js

```diff
@@ -266,3 +266,34 @@ export function cloneElement(adapter, el, props) {
     { ...el.props, ...props },
   );
 }
+
+export function spyMethod(instance, methodName) {
+  let lastReturnValue;
+  const originalMethod = instance[methodName];
+  const hasOwn = Object.hasOwnProperty.call(instance, methodName);
+  Object.defineProperty(instance, methodName, {
+    configurable: true,
+    enumerable: false,
+    value(...args) {
+      const result = originalMethod.apply(this, args);
+      lastReturnValue = result;
+      return result;
+    },
+  });
+  return {
+    restore() {
+      if (hasOwn) {
+        /* eslint-disable no-param-reassign */
+        instance[methodName] = originalMethod;
+        /* eslint-enable no-param-reassign */
+      } else {
+        /* eslint-disable no-param-reassign */
+        delete instance[methodName];
+        /* eslint-enable no-param-reassign */
+      }
+    },
+    get lastReturnValue() {
```

Comment:
I don't think this needs to be, or should be, a getter - can it just be a function?